src/js/legacy/mega-menu.js

```diff
@@ -38,8 +39,16 @@ class MegaMenu {
     this.openControl.addEventListener('click', this.showMenu);
     this.closeControl.addEventListener('click', this.hideMenu);
 
+    document.addEventListener('click', this.handleDocumentClick);
+    document.addEventListener('touchend', this.handleDocumentClick);
     window.addEventListener('resize', this.resetMenu);
+  }
 
+  handleDocumentClick(event){
+    const target = event.target;
+    if (!target.classList.contains('vetnav-level1')){
```

Comment:
Upon testing this out, if you click on any of the space not covered by links in the nav menus, you still close the menu. I don't think this is expected behavior.

Reply:
Also nitpick: space before the `{`.

Reply:
Good call! Previously there wasn't an event listener there at all so the menu stayed open unless you clicked the toggle button again. But now with an event listener on the document that closes the menu, there needs to be another event listener on the actual dropdown menu to keep that event from bubbling up. Working on it now, thanks!